File: 2°-Periodo/AEDs2/Estudos/BubbleSort.java
import java.util.*;
public class BubbleSort {

    public static void preencher(int array[], int n, Scanner sc){
        for(int i=0; i<n; i++){
            array[i] = sc.nextInt();
        }
    }

    public static void swap (int array[], int j){
        int aux = array[j];
        array[j] = array[j+1];
        array[j+1] = aux;
    }

    public static void bubbleSort(int array[], int n){
        for(int i=n-1; i>0; i--){
            for(int j=0; j<i; j++){
                if(array[j]>array[j+1]){
                    swap(array, j);
                }
            }
        }
    }

    public static void mostrar(int array[], int n){
        System.out.print("[ ");
        for(int i=0; i<n; i++){
            System.out.print(array[i] + " ");
        }
        System.out.println("]");
    }

    public static void main(String[] args){
        Scanner sc = new Scanner (System.in);

        int n = sc.nextInt();
        int[] array = new int[n];

        preencher(array, n, sc);

        bubbleSort(array, n);

        mostrar(array, n);

        sc.close();
    }
}

File: 2°-Periodo/AEDs2/Estudos/bubbleSort.c
#include <stdio.h>

void preencheArray(int array[], int n){
    for(int i=0; i<n; i++){
        scanf("%d", &array[i]);
    }
}

void swap(int array[], int j){
    int aux = array[j];
    array[j] = array[j+1];
    array[j+1] = aux;
}

void bubbleSort(int array[], int n){
    for(int i=n-1; i>0; i--){
        for(int j=0; j<i; j++){
            if(array[j]>array[j+1]){
                swap(array, j);
            }
        }
    }
}

void mostrar(int array[], int n){
    printf(" [");
    for(int i=0; i<n; i++){
        printf("%d ", array[i]);
    }
    printf("]");
}

int main(){
    int n;
    scanf("%d", &n);
    int array[n];

    preencheArray(array, n);

    bubbleSort(array, n);

    mostrar(array, n);
}